.gitignore

```diff
@@ -0,0 +1,34 @@
+bin/run-tests-config.sh
+.cache
+.ensime*
+.idea
+*.iml
+docker-volumes
+target
+.reviewboardrc
+dependency-reduced-pom.xml
+node_modules
+.project
+.classpath
+.settings
+.Rproj.user
+start.sh
+.DS_Store
+*.ipr
+*.iws
+__pycache__
+.history
+native/
+sandboxes/
+sandboxes.tar.gz
+tests/performance/results
+ci.log
+ci.tar.gz
+*.jfr
+.python-version
+.pytest_cache
+tools/packager/*.deb
+tools/packager/*.rpm
+tools/packager/toor
+tools/packager/fpm-docker/.provisioned
+dcos-registry-*
```

Comment:
- `dcos-registry-*` is definitely not needed :) 
- it would be nice to have the ignores can be categorized (like https://github.com/mesosphere/dcos-commons/blob/master/.gitignore may be)